src/cloud-api-adaptor/podvm/files/usr/local/bin/setup-scratch-storage

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+[ -n "${DEBUG:-}" ] && set -o xtrace
+
+handle_error() {
+    local exit_code="${?}"
+    local line_number="${1:-}"
+    echo "error:"
+    echo "Failed at $line_number: ${BASH_COMMAND}"
+    exit "${exit_code}"
+}
+trap 'handle_error $LINENO' ERR
+
+die() {
+    local msg="$*"
+    echo >&2 "ERROR: $msg"
+    exit 1
+}
+
+# Check if daemon.json exists and parse configuration
+DAEMON_CONFIG_PATH="/run/peerpod/daemon.json"
+SCRATCH_ENABLED="false"
+ENCRYPTION_ENABLED="false"
+
+if [ -f "$DAEMON_CONFIG_PATH" ]; then
+    if command -v jq &>/dev/null; then
+        SCRATCH_ENABLED=$(jq -r '.["enable-scratch-disk"] // false' "$DAEMON_CONFIG_PATH")
+        ENCRYPTION_ENABLED=$(jq -r '.["enable-scratch-encryption"] // false' "$DAEMON_CONFIG_PATH")
+    else
```

Comment:
if the code works with grep, why install jq?

Reply:
Right. looks like grep is default and should be there..